airflow/utils/log/es_task_handler.py

```diff
@@ -255,7 +256,9 @@ def close(self):
 
         # Mark the end of file using end of log mark,
         # so we know where to stop while auto-tailing.
-        self.handler.stream.write(self.end_of_log_mark)
+        if self.write_stdout:
+            print()
```

Comment:
What's this for?

Reply:
it is import for us to find the log_id mark in the log line with "end_of_log_mark" from the elastic search cluster. What I observed was that I saw the end of log mark can end up with the same line of the previous log lines hence it would prevent us from finding the end-of-log mark in some cases console prints from random places without the newline. I am adding an obnoxious new line (print()) so as to guarantee that end-of-log mark is a separate log record. For any other log line it is actually benign to have two log lines to combine into one line in elastic search. Only the end-of-log mark absolutely need to be in its own line.  This is just to make the solution here more robust and decoupled from the rest of log lines. I understand this is a fix for reliability and it probably is not very clean.

Reply:
This is to make sure the "end-of-log" mark log record will always be in a separate line therefore a separate log record. as those like us work with elastic search for logging, for various reasons, it is not always easy to ensure log lines are not combined. Because the end-of-log mark is so special, it helps with the print() here to guarantee that we can always find the mark with the log-id from es_task_handler.py.

Reply:
This needs a comment saying what it's for then please.